### ISSUE TITLE
使用next来迭代数据集时，张量形状发生改变

### ISSUE THREAD

**Issue by abcdefghHIM** [PaddlePaddle/Paddle#35053]
PaddlePaddle版本：2.1.2
自定义的Dataset片段：
class AudioDataset(Dataset):
..............
    def __getitem__(self, idx):
        p = self.files[idx]
        filename = os.path.basename(p)
        speaker = filename.split(sep='_', maxsplit=1)[0]
        label = self.encoder.transform([speaker])[0]
        mcep = np.load(p)
        mcep = paddle.to_tensor(mcep, dtype='float32')
        mcep = paddle.unsqueeze(mcep, 0)
        print(paddle.to_tensor(speakers.index(speaker), dtype='int64'))
        return mcep, paddle.to_tensor(speakers.index(speaker), dtype='int64'), paddle.to_tensor(label, dtype='float32')

测试：
def data_loader(datadir: str, batch_size=4, shuffle=True, num_workers=2):
    dataset = AudioDataset(datadir)
    loader = DataLoader(dataset, batch_size=batch_size, shuffle=shuffle, num_workers=num_workers)
    return loader
dloader = data_loader(config.data_dir, batch_size=config.batch_size, num_workers=config.num_workers)
data_iter = iter(dloader)
x,index,label=next(data_iter)
print('Test')
print(index)

输出：
Tensor(shape=[1], dtype=int64, place=CPUPlace, stop_gradient=True,
       [3])
Tensor(shape=[1], dtype=int64, place=CPUPlace, stop_gradient=True,
       [2])
Test
Tensor(shape=[1, 1], dtype=int64, place=CPUPlace, stop_gradient=True,
       [[3]])
Tensor(shape=[1], dtype=int64, place=CPUPlace, stop_gradient=True,
       [1])

**Comment by zhwesky2010** [comment]
@abcdefghHIM 您好，to_tensor只有在做将scalar转换为Tensor时，会将其改变为shape为[1]，请问是否存在这种情况呢？另外在dataset的def \__getitem\__函数中可以直接返回numpy，不用提前去做转换，dataloader迭代出的会自动转为Tensor
